src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = getDataFromFile(fromFileName);
+        String textToWrite = dataResearch(dataFromFile);
+        writeDataToFile(textToWrite, toFileName);
+    }
+
+    public String getDataFromFile(String file) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            StringBuilder builder = new StringBuilder();
+            String line = bufferedReader.readLine();
+            while (line != null) {
+                builder.append(line).append(" ");
+                line = bufferedReader.readLine();
+            }
+
+            return builder.toString();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file", e);
+        }
+    }
+
+    public String dataResearch(String text) {
+        int countOfSupply = 0;
+        int countOfBuy = 0;
+        StringBuilder stringBuilder = new StringBuilder();
+
+        String[] splitStr = text.split("\\s+");
+        for (int i = 0; i < splitStr.length; i++) {
+            String[] strArray = splitStr[i].split(",");
+            if (strArray[0].equals("supply")) {
+                countOfSupply += Integer.parseInt(strArray[1]);
+            } else {
+                countOfBuy += Integer.parseInt(strArray[1]);
```

Comment:
The code assumes that `strArray[1]` is always a valid integer and that the input data is well-formed. This could lead to `NumberFormatException` if the data is not as expected. Consider adding validation to check if `strArray[1]` is a valid integer before parsing.